src/frequenz/client/microgrid/_client.py

```diff
@@ -266,11 +266,11 @@ async def _new_component_data_receiver(
             broadcaster = streaming.GrpcStreamBroadcaster(
                 f"raw-component-data-{component_id}",
                 # We need to cast here because grpc says StreamComponentData is
-                # a grpc.CallIterator[PbComponentData], not a
-                # grpc.aio.UnaryStreamCall[..., PbComponentData].
-                lambda: cast(
-                    grpc.aio.UnaryStreamCall[Any, PbComponentData],
-                    self.api.StreamComponentData(PbComponentIdParam(id=component_id)),
+                # a grpc.CallIterator[PbComponentData] which is not an AsyncIterator,
+                # but it is a grpc.aio.UnaryStreamCall[..., PbComponentData], which it
+                # is.
+                lambda: self.api.StreamComponentData(
+                    PbComponentIdParam(id=component_id)
```

Comment:
The tests/checks failed, maybe the cast is missing?

From the tests/checks logs: `src/frequenz/client/microgrid/_client.py:8:1: F401 'collections.abc.AsyncIterator' imported but unused`

Reply:
Yeah, sorry, I just realized that I was trying out if the cast wasn't necessary anymore, but it is, I just updated again, hopefully this time is the last time :D